### PR TITLE
[Week4] - chaeryeong

### PR DESCRIPTION
**Pull request by fud21**
[ 심층 신경망(DNN) ]
- 심층 신경망에서의 문제점: 그래디언트 소실/폭주 문제, 학습 시간 증가, 오버피팅
- 활성화 함수: 시그모이드 (0, 1), tanh (-1, 1), ReLU 음수값을 0
- 가중치 초기화: Xavier 초기화, He 초기화
- 배치 정규화(Batch Normalization): 각 층의 활성화 값을 평균 0, 분산 1로 정규화.

[ 인공 신경망 (Artificial Neural Networks, ANN) ]
- 다층 퍼셉트론 : 입력층 → 은닉층 → 출력층
- 역전파 알고리즘 : 경사 하강법(Gradient Descent)을 통해 손실을 최소화.
- 신경망 설계의 주요 요소
은닉층의 수와 뉴런 수, 학습률, 배치 크기
- Keras API, Sequential API, 함수형 API, 서브클래싱 API
- 하이퍼파라미터 튜닝 : 랜덤 서치, 그리드 서치, 케라스 튜너
- 배치 정규화 (Batch Normalization)

과제1 : 배치 정규화를 포함한 딥러닝 모델 정의 함수
과제2 : 손글씨 숫자 분류기 만들기 : MNIST 데이터셋으로 다층 퍼셉트론(MLP)을 설계하여 손글씨 숫자 분류

### CODE REVIEW

**Review by kimhamyong — APPROVED** [review]
task1: 잘 수행하셨습니다.

**Comment by kimhamyong** [comment]
task1:

현재 케라스에선 `Dense(256, input_shape=(784,))` 대신 `Input` 레이어를 사용하는 것이 더 권장되고 있습니다.
   ```python
   from tensorflow.keras.layers import Input
   model = Sequential([
       Input(shape=(784,)),
       Dense(256),
       ...
   ])
   ```

**Review by giyomdabin — APPROVED** [review]
task2: 주어진 과제에 대해 잘 수행하셨습니다. 수고하셨습니다!
